Hide enum types in the stdlib and the manual (#644)
tweag/nickel: doc/manual/syntax.md

  return res"
```

#### Enum tags

Enumeration tags are used to express finite alternatives. They are formed
by writing a backtick `` ` `` followed by any valid identifier. For example,
`builtin.serialize` takes an export format as a first argument, which is an enum
tag among `` `Json ``, `` `Toml `` or `` `Yaml `` (as of version 0.1):

```nickel
builtin.serialize `Json {foo = 1}
# gives "{
#          \"foo\": 1
#        }"
builtin.serialize `Toml {foo = 1}
# gives "foo = 1
#       "
```

An enum tag `` `foo `` is serialized as the string `"foo"`:

```nickel
let as_yaml_string = builtin.serialize `Yaml {foo = `bar}
# gives "---
#       foo: bar
#       "

```

While it's technically possible to just use strings in place of enum tags, using
an enum tag insists on the fact that only a finite number of alternatives can be
used for the corresponding value.

Enum are also handled more finely by the typechecker, however, as of the version

> 0.1 of Nickel, enum types have been temporary disabled for incidental reasons.

Reviewer: Be more specific: “there are only two enum types[…]”

Author: You mean, the one from the stdlib? Those are standard custom contracts. I wrote them using enum types just because I can, because it's more concise and doesn't cost anything, but any user can write:

```nickel
MyComparisonResult = contract.from_predicate (fun value => value == `Less || value == `Equal || value == `Greater)
```

Should we really mention them as "enum types" in the doc, although they are just opaque types, statically?

Reviewer: I changed my mind, now that we've discussed the details. Let's say

```suggestion
Enum will also handled more finely by the typechecker, however, as of the version
0.1 of Nickel, enum types are not yet supported.
```
